apicast/src/provider.lua

```diff
@@ -385,8 +385,15 @@ function _M.access(service)
     ngx.var.cached_key = concat({service.id, params.app_id, params.app_key}, ':')
 
   elseif backend_version == 'oauth' then
-    ngx.var.access_token = parameters.access_token
-    params.access_token = parameters.access_token
+    if credentials.location == 'headers' then
```

Comment:
To clarify: 
- for user_key mode apicast would check both `user_key` (or whatever specified in the UI) AND `Authorization: Bearer` header? Or `user_key` query/body parameter and the `Authorization: Bearer` header?
- for app_id/app_key similar, but passing both id and key in the `Authorization: Bearer` header? (for example `<app_id>:<app_key>`)
- for OAuth check both `access_token` and `Authorization` header?

I think I would prefer to use `Authorization` header only for OAuth mode for now. But if you feel like other modes should also use it, I can make the change.